OpenRA.Mods.Common/Widgets/Logic/Settings/DisplaySettingsLogic.cs

```diff
@@ -269,16 +269,14 @@ Func<bool> InitPanel(Widget panel)
 			};
 
 			var colorManager = modData.DefaultRules.Actors[SystemActors.World].TraitInfo<ColorPickerManagerInfo>();
-			colorManager.Color = ps.Color;
 
-			var colorDropdown = panel.Get<DropDownButtonWidget>("PLAYERCOLOR");
-			colorDropdown.IsDisabled = () => worldRenderer.World.Type != WorldType.Shellmap;
-			colorDropdown.OnMouseDown = _ => ColorPickerLogic.ShowColorDropDown(colorDropdown, colorManager, worldRenderer, () =>
-			{
-				Game.Settings.Player.Color = colorManager.Color;
-				Game.Settings.Save();
-			});
-			colorDropdown.Get<ColorBlockWidget>("COLORBLOCK").GetColor = () => ps.Color;
+			SetColorWidget(panel, colorManager, () => worldRenderer.World.Type != WorldType.Shellmap, "PLAYERCOLOR", Game.Settings.Player, "Color");
+
+			bool IsStanceDisabled() => !Game.Settings.Game.UsePlayerStanceColors && worldRenderer.World.Type != WorldType.Shellmap;
```

Comment:
```suggestion
			bool IsStanceDisabled() => !Game.Settings.Game.UsePlayerStanceColors;
```

Why is this always enabled on the shell map?

Reply:
because you don't want the user to edit while ingame. We should make it dynamic in the future, but that needs more plumbing and I'm not sure how should that look like.

Reply:
To me: when the feature is disabled: the configuration is greyed out. Always.

Reply:
as it should, it's a useless option if these colors aren't used